### ISSUE TITLE
Since v2.0, app no longer appears in android share menu

### ISSUE THREAD

**Issue by malmsey** [standardnotes/mobile#11]
This is on android 7.1.  Love the update, except that I really need this one feature...

**Comment by mobitar** [comment]
Hey, thanks, hoping to get this back up soon. Can you describe how you used this feature btw?

**Comment by malmsey** [comment]
I think I've only tested it when sharing URLs from the Slide for Reddit app.  The new Standard Notes is nowhere to be seen on the list of apps that pops up.

**Comment by dwu** [comment]
Because you asked for descriptions of how this feature is used, here's a brief summary of how I use it.

I use standardnotes mainly as a tool to quickly write down notes and capture ideas while on mobile and sort/work on them later on another device. Standardnotes' instant and robust synchronizsation makes it perfect for this use case in my opinion.

Contentwise I mostly capture URLs of articles and short snippets of text from either the browser, a feed reader or a PDF reader. I typically tag the shared content as #readme and/or the name of the project the content relates to.

As there's currently no way to share from applications directly to standardnotes, sharing involves copying the content to be shared to the clipboard, finding and switching to standardnotes via the task manager, manually creating a new note and pasting the content.

**Comment by mobitar** [comment]
Got it, thanks!

**Comment by foorb** [comment]
Hi, same on iOS (11.2.6). Not sure if it ever worked before, but it would be nice to have.

**Comment by mobitar** [comment]
This never made it on iOS ;) But indeed, would be nice to have.

**Comment by ni-ka** [comment]
+1 for bringing this back. used it quite a lot with evernote, saves some time and avoids having to switch context.

**Comment by malmsey** [comment]
Yes, it was a real time saver when you could do it in 1.0.  When you shared a URL to standard notes the url would automatically be copied into the body of a new note and there would usually be some kind of autofill in the title field as well.  I hope this feature can be brought back!

**Comment by klevstul** [comment]
+1 as i need this feature

**Comment by isaacsu** [comment]
+1 for this too, please.

I'd like to use Standard Notes it to "bookmark" links from iOS Safari that I want to come back to. At the moment it's one step Trello/Evernote/Signal-to-self, or four step Copy, Open Standard Notes, New Note, Paste.

**Comment by mobitar** [comment]
While not exactly it, I have long considered a setting for mobile titled "Open new note composer on app launch", which would automatically open a new note when you launch the app, making quick adding simpler. Would that be useful?

**Comment by malmsey** [comment]
I don't know about iOS but before the 2.0 rewrite, standard notes android could do that in one step (not 4 like now, or possibly 3 with the above option).  What happened to the share functionality?

**Comment by 1337** [comment]
Re "Open new note composer on app launch", you are right it's not exactly it for Android, but it might still be useful for iOS.

With the new note composer setting, my workflow for saving a link or snippet will be as such:
1. select URL to copy (2 taps)
1. go home (1 tap)
1. select standard notes (1 or 2 taps)
1. paste in contents (2 taps)
1. go back to to the browser/app (2 taps)

With an actual share functionality, the number of taps is halved:
1. tap on [chrome or whatever browser] menu (0 or 1 tap, depending on browser)
1. choose share to standard notes (1 or 2 taps)
1. tap Done (1 tap)

So yes, there is still merit to implementing proper share functionality for a use case such as mine.

**Comment by isaacsu** [comment]
> "Open new note composer on app launch"
Not really @mobitar because then I'll have to dismiss the note compose whenever I want to just read a note.

**Comment by mobitar** [comment]
The rewrite in React Native made implementing sharing into the app a little tricky, so it was postponed. Also, wanted to prove app stability before implementing something like this. Generally I will say that I adopt the mindset of not being too rushed on any feature request. Otherwise if I acted too quickly on every request, it would open the floodgates to bloat. Things will _always_ be one or two extra steps with SN. That's a feature. Every line of code creates exponential maintenance debt. So you give up a tiny bit of convenience for improved stability and longevity, which is our main tenet.

**Comment by MayhemYDG** [comment]
I'd really like for this feature to be implemented, it's the last thing that I need before I completely jump ship from Google Keep.
FWIW the share implementation on Keep is really nice, it only brings up a little pop-up window, not the entire app.

**Comment by mobitar** [comment]
This isn't just a matter of philosophy: adding this feature right now and having to maintain it and test it on several devices, now and for the foreseeable future, would cause a huge resource strain. I'm ok with losing users because of it if it means maintaining simplicity and maintainability. I understand its use case and will probably add it at some point, but just can't make any promises right now.

**Comment by daniele-athome** [comment]
Wonderful app guys, but this is a blocking thing. Seriously, sharing between apps is one of the things that makes a mobile platform very powerful. You're locking yourself out of the ecosystem this way. I know there are other things to do, but keep this in mind.

Other than that, fantastic application!

**Comment by mobitar** [comment]
Definitely in mind! Hopefully soon.

**Comment by robot3498712** [comment]
Hi, I am looking into enhancing my workflow by using cross-platform notes/sync apps, so this, agreeing with comments above, absolutely is a blocker for me (main motivation for bothering is system integration without writing the apps myself). Simplenote for example can do it, alas is lacking other features like encryption or self-hosting.
Thanks for your efforts and product :)

**Comment by RavetcoFX** [comment]
@mobitar would you be willing to merge this if I made a pr for this and maintained it?

**Comment by mobitar** [comment]
There's actually a pending PR for this already by @jh-code: https://github.com/standardnotes/mobile/pull/133

I just haven't had the chance to review it yet as I've been prioritizing bug-fix releases over feature releases. But I'm thinking if it works well, we can get it in within the next few releases.

**Comment by mobitar** [comment]
But if you @RavetcoFX or anyone else wants to test out the PR and confirm it works well, that would help instill more confidence for me in the PR and potentially get it merged sooner.

**Comment by yield65** [comment]
Hello,
Is there any progress on this? Because that's the main feature i need to work between mobile (phone/tablet) and office laptop.

**Comment by JaspalSuri** [comment]
I'm afraid we can't comment on its progress, but I've passed this on to the team for review.

**Comment by lifofernandez** [comment]
This is a must, coming from other note keeping app, it's a basic feature every content management app already has.

**Comment by lifofernandez** [comment]
This is a very basic feature for a content managing/authoring app.

**Comment by JaspalSuri** [comment]
I understand. After the rewrite, we can work on re-adding this feature and/or accept any usable PRs.

**Comment by OllySa** [comment]
Absolute face-palm when in another app I went to 'Share' to make a new note (Android) ... there in the Share menu is Google Keep, Joplin and others, but I scroll up and down in utter disbelief that Standard Notes isn't there in the share menu. Guys.

Use case: I'm in the browser or my podcast app, or any of many other apps, and want to 'share' the content/URL to a new note. Then I then tend to add some tags and thoughts to the note. I admire your philosophy, but is this really not considered 'basic' functionality? For me it is.. and it seems I'm not alone.

My intention is to pay you guys for the 5 year plan. But of course I can't do that until it works for my basic use cases, including: making a new note by sharing from another app, or opening the app to quickly make a note (this second being perhaps an even bigger showstopper, as when I open the app it currently takes up to 30seconds (!) before I can do anything. But I've mentioned that [elsewhere](https://github.com/standardnotes/mobile/issues/240#issuecomment-712722028) - and thanks for your response, although no solution).

Sorry for my frustrated tone, but I am - I've been waiting years to move to Standard Notes now. Only being harsh because I care - I think yours could be the best product out there (and help me de-monopoly my digital life just a little ;-) 

I just really don't want to find that 3.5 doesn't add/fix these.

**Comment by mobitar** [comment]
No need to share the use case, totally understand this is important. 3.5 doesn't add new features; instead, it's a robust foundation on which we can build new features (like this one). But adding sharing functionality in 3.x is definitely a priority (meaning it's something we hope to get to far before 4.0).

cc @radko93

**Comment by introspectionism** [comment]
I’ve been waiting for this feature for a long time, as I wanted to go full in using Standard Notes. I can’t fathom why this feature is lacking. It should be trivial to implement. Or, what am I missing here?

I will not renew my Standard Note subscription unless this gets fixed. This issue has been open for over four years. No progress.

**Comment by zosoguidi** [comment]
Indeed, I'm also desperate and have been waiting quite a while to be able to share content into my standardnotes app from another app instead of clumsily copy-pasting. 

I have no idea how difficult or easy it is to do this; specially considering SNs fantastic security. But given how excellent the platform is otherwise I'm hoping the delay in reintroducing this feature is because they are taking their time to do it properly and to their standards........
